app/models/auth_token.rb

```diff
@@ -13,10 +13,26 @@ def data
 
 private
 
-  attr_reader :jwt_token
+  attr_reader :token
 
   def read_token
-    payload, = JWT.decode(jwt_token, secret, true, algorithm: "HS256")
+    read_message_encryptor_token || read_jwt_token
+  end
+
+  def read_message_encryptor_token
+    len = ActiveSupport::MessageEncryptor.key_len
+    key = ActiveSupport::KeyGenerator.new(secret).generate_key("", len)
```

Comment:
Are we sure there are no risks or clues given if we don't have a salt? we could always add one as an environment variable?

Reply:
If we specify a salt in an environment variable, would we have to change it frequently?  Does a static salt add much security over a nonexistent salt?

Reply:
I don't personally really understand the role of salt in symmetric encryption 🤔 It makes a lot of sense in hashing.

Reply:
Looks like [ActiveSupport::KeyGenerator uses PBKD2F](https://api.rubyonrails.org/classes/ActiveSupport/KeyGenerator.html), and the role of the salt there is [to prevent rainbow table attacks to reverse-engineer the password](https://crypto.stackexchange.com/questions/3484/pbkdf2-and-salt)

Reply:
Oh I see - so it's used on a one way hash on the key. Thanks 👍. I can understand why it may be of no value in this situation.